config.go

```diff
@@ -174,9 +174,15 @@ func getPluginVersion() string {
 	return "unknown"
 }
 
-func isValidIAM(cident *string) bool {
-
-	if strings.Contains(*cident, "assumed-role/Admin/") || strings.Contains(*cident, "assumed-role/IAMAdmin/") {
+func isValidIAM(arn *string) bool {
+
+	/*
+		Validates ARN for assumed-role of:
+			- Admin
+			- IAMAdmin
+			-Machine Identities.
+	*/
+	if strings.Contains(*arn, "assumed-role/Admin/") || strings.Contains(*arn, "assumed-role/IAMAdmin/") || strings.Contains(*arn, "arn:aws:sts::") {
```

Comment:
This won't be sufficient  as we also allow STS credentials for interactive users (aka ALKS credentials) and this would match other role types that are not IAM enabled.  Going to need to do a bit of brainstorming on this.

Reply:
I _think_ that a MI should be able to call `/rest/roleMachineIdentity/search` for itself and receive a valid response.  We'd need to do some testing to confirm, but that's probably the best way to validate that a non-Admin/IAMAdmin role session is actually a MI.  Perhaps we could pre-filter the ARN for `acct-managed` first to weed out any other interactive users before we make another API call.

Reply:
ohh that is a good idea. let me string together some code and see if I can get this to work.